x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/persistence/TrainedModelProvider.java

```diff
@@ -550,7 +551,9 @@ private InferenceStats handleMultiNodeStatsResponse(SearchResponse response, Str
             failures == null ? 0L : Double.valueOf(failures.getValue()).longValue(),
             modelId,
             null,
-            timeStamp == null ? Instant.now() : Instant.ofEpochMilli(Double.valueOf(timeStamp.getValue()).longValue())
+            timeStamp == null || (Numbers.isValidDouble(timeStamp.getValue()) == false) ?
```

Comment:
Is it possible for timestamp to be an invalid number here?

Reply:
Yes, it is. The agg could have not found any values, or the values they found were changed somehow. This type of check is effectively what aggs do under the hood when they serialize to JSON to protect against goofy doubles. The difference here is that we write out `now()` instead of `null`